src/api/services/apiKeyService.ts

```diff
@@ -25,13 +27,21 @@ export const getApiRoles = async (participant: Participant) => {
 
 export const validateApiRoles = async (
   keyRoles: string[],
-  participant: Participant
+  allowedRoles: ApiRoleDTO[]
 ): Promise<boolean> => {
-  const participantRoles = (await getApiRoles(participant)).map((role) => role.roleName);
+  const participantRoles = allowedRoles.map((role) => role.roleName);
 
   for (const role of keyRoles) {
     if (!participantRoles.includes(role)) return false;
   }
 
   return true;
 };
+
+export const getApiKey = async (siteId: number, keyId: String): Promise<ApiKeyDTO | undefined> => {
+  const apiKeyAdmin = await getApiKeyFromAdmin(keyId);
+
+  if (apiKeyAdmin?.site_id !== siteId) return undefined;
```

Comment:
Do you see any point in extracting this conditional to a variable, say:
```
const isApiKeyAssociatedWithSite = apiKeyAdmin?.site_id === siteId;
```
I think it would make it clearer

Reply:
Ah yeah that makes sense!